tests/unit/api/test_upload_sessions.py

```diff
@@ -113,6 +113,32 @@ async def test_get_session_status(
             workspace_name="sdk_read", endpoint=f"upload_sessions/{session_id}"
         )
 
+    async def test_get_session_status_with_retry(
```

Comment:
we should also add a test that tests the case where every response returns a 502